pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -215,6 +215,12 @@ func AddToManager(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	// Watch for changes to SyncSetInstance
+	err = c.Watch(&source.Kind{Type: &hivev1.SyncSetInstance{}}, &handler.EnqueueRequestForObject{})
```

Comment:
you want to queue up a ClusterDeployment here instead of the SyncSetInstance object